File: SMSboxunemp/SMSboxunemp.R
rm(list=ls(all=TRUE))                                       # remove variables
graphics.off()                                              # close all windows
# setwd("C/...")                                            # set working direktory

load("unemploy.rda")                                        # load data

x = unemploy[,2]                                            # total unemployment
e = c( 15.1, 15.8, 16.8, 17.1, 17.3, 19.9);                 # unemployment East Germany
w = c( 5.8, 6.2, 7.7, 7.9, 8.7, 9.8, 9.8, 9.8, 10.4, 13.9); # unemployment West Germany

 # calculating means
means = c(mean(x), mean(e), mean(w));
 # producing boxplot without means
boxplot(x,e,w, names = c("All", "East", "West"), main = "Unemployment in Germany");
 # adding means into boxplot
lines(x=c(0.6,1.4),rep(means[1],2),lty=2)
lines(x=c(1.6,2.4),rep(means[2],2),lty=2)
lines(x=c(2.6,3.4),rep(means[3],2),lty=2)

File: SMSboxunemp/SMSboxunemp.m
 % clear history
clear all
close all
clc

 % load data
load unemploy.dat

 % preparing data
all = unemploy;
we = [3 3 2 3 3 3 3 3 3 3 3 2 2 2 2 2]'; % 3=West, 2=East

j=1;
l=1;
for i=1:length(we)
    if (we(i)==3)
        w(j) = all(i); % west
        j = j+1;
    elseif (we(i)==2)
        e(l) = all(i); % east
        l = l+1;
    end
end

 % calculating means
means = [mean(all), mean(e), mean(w)];


 % resampling for plot
data = [all;all];
groups = [ones(length(all),1);we];

hold on
 % adding means into plot
line([0.775 1.225],[means(1) means(1)],'Color','k','LineStyle',':','LineWidth',1.2)
line([1.775 2.225],[means(2) means(2)],'Color','k','LineStyle',':','LineWidth',1.2)
line([2.775 3.225],[means(3) means(3)],'Color','k','LineStyle',':','LineWidth',1.2)
 % producing boxplots
boxplot(data,groups,'Symbol','o', 'labels',{'All', 'East', 'West'});
title('Unemployment in Germany','FontSize',16,'FontWeight','Bold');
set(findobj(gca,'Type','text'),'FontSize',16,'FontWeight','Bold');
box on
set(gca,'LineWidth',1.6,'FontSize',16,'FontWeight','Bold')   % set frame
hold off


 % to save plot please uncomment following lines 
 % print -painters -dpng -r600 SMSboxunemp.png
 % print -painters -dpdf -r600 SMSboxunemp.pdf